codecs/resize/src/lib.rs

```diff
@@ -36,7 +37,17 @@ pub fn resize(
         3 => Type::Lanczos3,
         _ => panic!("Nope"),
     };
+    let num_input_pixels = input_width * input_height;
     let num_output_pixels = output_width * output_height;
+
+    if premultiply {
+      for i in 0..num_input_pixels {
+        for j in 0..3 {
+          input_image[4 * i + j] = ((input_image[4 * i + j] as f32) * (input_image[4 * i + 3] as f32)/255.0) as u8;
```

Comment:
The int->float->int conversion isn't technically necessary (you can use `u16` instead).